feat: add support for the typed array `every` method in `array/complex64` (#1184)
stdlib-js/stdlib: lib/node_modules/@stdlib/array/complex64/README.md

// returns undefined
```

<a name="method-every"></a>

#### Complex64Array.prototype.every( predicate\[, thisArg] )

> Returns a boolean indicating whether all elements pass a test function.

Reviewer: ```suggestion
Returns a boolean indicating whether all elements pass a test.
```

```javascript
var realf = require( '@stdlib/complex/realf' );
var imagf = require( '@stdlib/complex/imagf' );

function predicate( v ) {
    return ( realf( v ) === imagf( v ) );
}

var arr = new Complex64Array( 3 );

// Set the first three elements:
arr.set( [ 1.0, 1.0 ], 0 );
arr.set( [ 2.0, 2.0 ], 1 );
arr.set( [ 3.0, 3.0 ], 2 );

// Check whether all elements pass a test:
var z = arr.every( predicate );
// returns true
```

> The `predicate` function is provided 4 arguments:

Reviewer: ```suggestion
The `predicate` function is provided four arguments:
```

// returns undefined
```

<a name="method-every"></a>

#### Complex64Array.prototype.every( predicate\[, thisArg] )

> Returns a  boolean indicating whether all elements pass a test function predicate or not.

Reviewer: ```suggestion
Returns a boolean indicating whether all elements pass a test function.
```

```javascript
var realf = require( '@stdlib/complex/realf' );
var imagf = require( '@stdlib/complex/imagf' );

> var arr = new Complex64Array( 3 );
> 
> // Test function
> function predicate( v ) {
>     return ( realf( v ) === imagf( v ) );
> }

Reviewer: ```suggestion
function predicate( v ) {
    return ( realf( v ) === imagf( v ) );
}

var arr = new Complex64Array( 3 );
```

> // Set the first, second, and third elements:

Reviewer: ```suggestion
// Set the first three elements:
```

arr.set( [ 1.0, 1.0 ], 0 );
arr.set( [ 2.0, 2.0 ], 1 );
arr.set( [ 3.0, 3.0 ], 2 );

> // Check whether all elements pass a test or not

Reviewer: ```suggestion
// Check whether all elements pass a test:
```

var z = arr.every( predicate );
// returns true
```

> The test function `predicate` is provided 4 arguments:

Reviewer: ```suggestion
The `predicate` function is provided 4 arguments:
```

> -   **thisArg**: execution context
> -   **value**: element of complexArray
> -   **index**: index of element
> -   **this**: ComplexArray prototype

Reviewer: ```suggestion
-   **value**: current array element.
-   **index**: current array element index.
-   **arr**: the array on which this method was called.
```

To set the function execution context, provide a `thisArg`.

```javascript

> // Test Function

Reviewer: ```suggestion
```

function predicate( v, i ) {

>     this.sum += 1;

Reviewer: ```suggestion
    this.count += 1;
```

>     return ( i >= 0);

Reviewer: ```suggestion
    return ( i >= 0 );
```

}

var arr = new Complex64Array( 3 );

var context = {

>     'sum': 0

Reviewer: ```suggestion
    'count': 0
```

Our convention is to use `count` for this.

};

> // Set the first, second, and third elements:

Reviewer: ```suggestion
// Set the first three elements:
```

arr.set( [ 1.0, 1.0 ], 0 );
arr.set( [ 2.0, 2.0 ], 1 );
arr.set( [ 3.0, 3.0 ], 2 );

var z = arr.every( predicate, context );
// returns true

> z = context.sum;

Reviewer: ```suggestion
var count = context.count;
```
